### PR TITLE
[don't merge] [MIR] Implement Torrent of Lava

### PR DESCRIPTION
**Pull request by Cguy7777**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/t/TorrentOfLava.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    public TorrentOfLava copy() {
+        return new TorrentOfLava(this);
+    }
+}
+
+class TorrentOfLavaGainAbilityEffect extends GainAbilityAllEffect {
+
+    TorrentOfLavaGainAbilityEffect() {
+        super(new SimpleActivatedAbility(new TorrentOfLavaPreventionEffect(null, null), new TapSourceCost()),
+                Duration.WhileOnBattlefield,
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/t/TorrentOfLava.java:64]: My gut feeling is that this should be `Duration.Custom`? But I haven't tested; if you've tested and it works then all good.

(this is a weird card, so if you feel up to writing a unit test that would be cool!)

Comment (Cguy7777) [reply Mage.Sets/src/mage/cards/t/TorrentOfLava.java]: Good point, `Duration.Custom` seems more appropriate.

**Comment by JayDi85** [comment]
Must create new PR:

IMG_9514

**Comment by Cguy7777** [comment]
My bad, I merged in changes for testing and forgot to fix it. I'll open a new PR.

**Pull request closed without merging**